src/sitemap/sitemaps.py

```diff
@@ -238,33 +239,39 @@ def __init__(self, region, language):
         # Instantiate WebappSitemap
         super().__init__(region, language)
         # Filter queryset based on region
-        self.queryset = self.queryset.filter(region=self.region)
+        self.queryset = self.queryset.filter(regions=self.region)
 
     def location(self, obj):
         """
         This location function returns the absolute path for a given object returned by items().
 
         :param obj: Objects passed from items() method
-        :type obj: ~cms.models.offers.offer.Offer
+        :type obj: ~cms.models.offers.offer_template.OfferTemplate
 
         :return: The absolute path of the given offer object
         :rtype: str
         """
-        return "/" + "/".join([obj.region.slug, self.language.slug, "offers", obj.slug])
+        return reverse(
+            "edit_offer_template",
+            kwargs={
+                "offer_template_id": obj.id,
+            },
+        )
```

Comment:
```suggestion
        return "/" + "/".join([obj.region.slug, self.language.slug, "offers", obj.slug])
```
The sitemap does not generate all urls available in the CMS, but all urls of the WebApp.
Thus, this should be the absolute url to the offer in the WebApp (e.g. https://integreat.app/augsburg/de/offers/sprungbrett), not to the edit form in the CMS.
I noticed that this doesn't work for offers which are immediately opened in an external website, but this is not part of this PR, I will open a new issue for it.

Reply:
Yes it was my fault to set the link to the CMS here, but I'm not sure, if that would work, because I called the relationship `regions`. As the the queryset is filtered to only OfferTemplates for a single region, we could use `obj.regions.first().slug` or simpler `self.region.slug`.
On the other side my intention to use reverse here was to not violate the [DRY principle](https://docs.djangoproject.com/en/3.1/topics/http/urls/#reverse-resolution-of-urls). So this hardcoded string would correspond to `reverse("api_offers", kwargs={"region_slug": self.region.slug, "language_slug": self.language.slug}) + obj.slug`.
What do you think about this, does DRY matter here?

Reply:
Yes, in general you're right, but we can't use the `reverse()` function for this, because links to the webapp are not defined in `urls.py` and will never be.
A different approach could be to use the function [get_absolute_url()](https://docs.djangoproject.com/en/3.1/ref/models/instances/#get-absolute-url) for all model instances and calculate the url to the webapp there. On the long term, we should also do this for pages, events, etc. but it's a good start to do this for offers now...

Reply:
And you're also right about the region parameter, just `obj.region` won't work. However, your solution doesn't work as well, because even if you have the queryset of all offer templates of region A, it could still be that `obj.regions.first()` is region B if it is activated in that region as well. So maybe we need a similar method to the ones in the API `get_url(offer, region)` which calculates the correct URL. And `get_absolute_url()` can't be used for this, because it doesn't have access to the region object. So I suggest to either leave the urls hardcoded in the sitemaps.py, or add a property `get_url(self, region)` to the offer template model...
Or do you have any other ideas?

Reply:
This ManyToManyField pitfalls :smile: 
So I like your idea about the offer template property. I will add a commit soon.

Reply:
Alright, this is done in d2b7a3e8ac442880d4c24f073106e1f028f42ad2, finally I just left it hardcoded, but just updated the region slug reference